src/playwrightTestCLI.ts

```diff
@@ -37,6 +39,22 @@ export class PlaywrightTestCLI {
   }
 
   reset() {
+    if (!this._traceViewers)
+      return;
+    for (const traceViewer of this._traceViewers)
+      traceViewer.dispose();
+    this._traceViewers = undefined;
+  }
+
+  async availableTraceViewers(): Promise<TraceViewer[]> {
```

Comment:
Why not traceViewer() that would return enabled instance? You don't seem to save much via having the enabled check on the call site. Also, if those are shallow objects, why not created them in the constructor? It would read better.

Reply:
I was doing it that way because I was only checking if the trace viewer supported the playwright version in the extension, so I needed to send both implementations.

I'm now changing that to ensure that we only request the enabled trace viewer when needed, and so checking the playwright version can be done here or in testModel, so that we can only pass the right trace viewer to the call site.